api-reference/10 UI Components/GridBase/1 Configuration/pager/allowedPageSizes.md

```diff
@@ -9,4 +9,4 @@ default: 'auto'
 Specifies the available page sizes in the page size selector.
 
 ---
-The UI component determines the allowed page sizes depending on the amount of data by default.
\ No newline at end of file
+The component processes the data to determine the page sizes available in the page size selector. Use this property to specify custom page sizes. If you want an additional option to display all rows on a single page, add *"all"* to the **allowedPageSizes** array.
```

Comment:
- "The component processes the data to determine the page sizes available in the page size selector." - this is unclear. Processes how? Based on what? What will be the result? No indication that this is the default behavior.
- "If you want an additional option to display all rows on a single page" - this option is meant for users, not for devs. I', also not sure that the word "option" is a good choice here because we used it extensively in a different sense not so long ago.
- "Use this **property**... add *"all"* to the **allowedPageSizes** **array**." - Although the property does accept the array, it confuses the reader when you use "property" and "array" as interchangeable terms without indicating that they are so.

```suggestion
Set this property to an array of numbers that specify available page sizes. If you want to let users display all records on a single page, add the *"all"* value to this array. The default value *"auto"* means that the component automatically determines available page sizes based on the total number of records.
```